akka-stream/src/main/scala/akka/stream/scaladsl/Tcp.scala

```diff
@@ -183,5 +183,5 @@ final class Tcp(system: ExtendedActorSystem) extends akka.actor.Extension {
    * It represents a prospective TCP client connection to the given endpoint.
    */
   def outgoingConnection(host: String, port: Int): Flow[ByteString, ByteString, Future[OutgoingConnection]] =
-    outgoingConnection(new InetSocketAddress(host, port))
+    outgoingConnection(InetSocketAddress.createUnresolved(host, port))
```

Comment:
Basic thing is that somewhere down the Chain the `TcpOutgoingConnection` actor is used.
This Actor checks if `remoteAddress.isUnresolved` if not it will send a `Dns.resolve` command to the `InetAddressDnsResolver`. This gives back the IP Address which then calls `register(new InetSocketAddress(resolve.addr, remoteAddress.getPort))` which then in fact opens the connection.
(Actually you can resolve the address yourself or let akka handle it)